src/mutation/RelayMutationQueue.js

```diff
@@ -133,6 +133,18 @@ class RelayMutationQueue {
 
   rollback(id: ClientMutationID): void {
     const transaction = this._get(id);
+    const collisionKey = transaction.getCollisionKey();
+    const collisionQueue =
+        collisionKey && this._collisionQueueMap[collisionKey];
+    if (collisionQueue) {
+      const index = collisionQueue.indexOf(transaction);
+      if (index !== -1) {
+        collisionQueue.splice(index, 1);
+      }
+      if (collisionKey && collisionQueue.length === 0) {
+        delete this._collisionQueueMap[collisionKey];
+      }
+    }
```

Comment:
@yungsters  Thanks a lot for feedback.
`collisionQueue` is managed in `commit`, `_advanceCollisionQueue`, `_failCollisionQueue`.  So If I move it to `_clearPendingTransaction` than both `_failCollisionQueue` and `_clearPendingTransaction` will try to remove it for certain code path (its necessary to slice() queue in `_failCollisionQueue` to iterate it correctly) from the `collisionQueue`.

And its not possible just move this responsibility only to `_clearPendingTransaction` since its necessary to clean `collisionQueue` also for situations when its commited again (not rollbacked).

But if you still prefer it I will move it there and adjust `_failCollisionQueue` to deal with that by using `slice` and iterating on copy:
```
  _failCollisionQueue(transaction: PendingTransaction): void {
    const collisionKey = transaction.getCollisionKey();
    if (collisionKey) {
      const collisionQueue = nullthrows(this._collisionQueueMap[collisionKey]);
      // Remove the transaction that called this function.
      collisionQueue.shift();
      collisionQueue.slice().forEach(
        transaction => this._handleCommitFailure(transaction, null)
      );
      delete this._collisionQueueMap[collisionKey];
    }
  }

```